### PR TITLE
Add V2 events to cookiecutter

### PR DESCRIPTION
**Pull request by spladug**

### CODE REVIEW

File: baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def make_wsgi_app(app_config):
 {% endif -%}
 {%- if cookiecutter.integrations.events %}
     baseplate.add_to_context("events_production", EventQueue("production"))
-    baseplate.add_to_context("events_test", EventQueue("test"))
+    baseplate.add_to_context("events_v2", EventQueue("v2", event_serializer=serialize_v2_event))
```
Comment (pacejackson) [inline baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: Why does it have both v1 and v2?

Comment (spladug) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: Hm, I wasn't sure if v2 was able to completely replace v1 yet. I'm very happy to delete it if so though.

Comment (krishnan-chandra) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: Naming here is a bit confusing. Is this v2 production or v2 test? I would assume the former but probably best to specify explicitly.

Comment (krishnan-chandra) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: To answer your other question, I don't believe we will be able to fully deprecate v1 for quite some time, so please leave it in.

Comment (pacejackson) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: For existing stuff sure, but this is for new services, do we have to add new events using the v1 format or can we encourage people to use v2 going forward?

Comment (spladug) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: It's "v2" not v2-prod _or_ v2-test as I'm trying to get rid of the idea of there being two different queues to write to and instead just vary where the publisher sends based on the environment the service is in (prod -> prod, staging -> test).

Will we be able to say "no new events on v1"? or do you anticipate new services doing new events in v1 for the forseeable future?

Comment (pacejackson) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: If we do have to keep v1 in the template, maybe we should rename it from `events_production` to `events_v1`?

Comment (spladug) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: That requires some puppet changes in prod to be compatible :grimacing: (I'm not opposed, but lazy)

Comment (pacejackson) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: 😅 

I don't feel strongly about it.

Comment (pacejackson) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: Wait... you need a puppet change to first argument in `add_to_context` from `"events_production"` to `"events_v1"`?  Or to rename the first argument to `EventQueue` from `"production"` to `"v1"`?

Comment (krishnan-chandra) [reply baseplate_cookiecutter/{{cookiecutter.project_slug}}/{{cookiecutter.module_name}}/__init__.py]: So, there are a few new events from v1 but those are from r2 and not baseplate services. I think it's fine to mandate that Baseplate services not send v1 events from now on.

**Comment by spladug** [comment]
:haircut_man:

**Comment by spladug** [comment]
69329bd9d6f2e5c1c20084ca75ab8b09b89909b7...9fc356be9262e2bf01e83e607a5785cb21e091b5

**Pull request closed without merging**